src/pages/events/create.tsx

```diff
@@ -0,0 +1,202 @@
+import {
+  Input,
+  FormLabel,
+  Checkbox,
+  CheckboxGroup,
+  Stack,
+  VStack,
+  Button,
+  useToast,
+  Text,
+} from '@chakra-ui/react'
+import { z } from 'zod'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { DataTable } from '../../components/events/DataTable'
+import { trpc } from '../../utils/trpc'
+import { useSession } from 'next-auth/react'
+import React, { useState } from 'react'
+import { useForm } from 'react-hook-form'
+import Head from 'next/head'
+import LoadingScreen from '~/components/common/LoadingScreen'
+import RestrictedPage from '~/components/auth/RestrictedPage'
+import Container from '~/components/auth/Container'
+import { useRouter } from 'next/router'
+
+const EventPage = () => {
+  const router = useRouter()
+  const toast = useToast()
+
+  const FormSchema = z.object({
+    eventName: z.string().min(1, { message: 'Invalid name' }),
+    dept: z
+      .array(
+        z.string({
+          required_error: 'At least one department must be chosen',
+          invalid_type_error: 'At least one department must be chosen',
+        })
+      )
+      .nonempty({
+        message: 'At least one department must be chosen',
+      }),
+    date: z.preprocess((arg) => {
+      if (typeof arg == 'string' || arg instanceof Date) return new Date(arg)
+    }, z.date().min(new Date(), { message: 'A valid date is required' })),
+  })
+
+  type FormSchemaType = z.infer<typeof FormSchema>
+
+  // useForm for state management except attendees which belongs in DataTable (child)
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+  } = useForm<FormSchemaType>({
+    resolver: zodResolver(FormSchema),
+  })
+
+  const [attendees, setAttendees] = useState<string[]>([])
+  const [submitBefore, setSubmitBefore] = useState<boolean>(false) // hacky use for attendees validation
+  const { data: session } = useSession()
+  const { data: data } = trpc.event.getAllUsers.useQuery()
```

Comment:
- [ ] No need `{data: data}`, destructuring `const {data} = useSession()` will suffice.
- [ ] Do add in a `useSession({ required: true })` as it will require the user to be logged in prior to accessing the page.